app/queries/session_query.rb

```diff
@@ -0,0 +1,47 @@
+class SessionQuery
```

Comment:
Is it necessary to have this Session Query? Seems to be a bit over-engineering

Also, I think here is a clearer way to define the query object. Removing the scopes from the model and passing them to the query object
https://thoughtbot.com/blog/a-case-for-query-objects-in-rails